src/Compilers/Core/MSBuildTask/InteractiveCompiler.cs

```diff
@@ -0,0 +1,328 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Linq;
+using Roslyn.Utilities;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    /// <summary>
+    /// This class defines all of the common stuff that is shared between the Vbc and Csc tasks.
+    /// This class is not instantiatable as a Task just by itself.
+    /// </summary>
+    public abstract class InteractiveCompiler : ToolTask
+    {
+        internal readonly PropertyDictionary _store = new PropertyDictionary();
+
+        public InteractiveCompiler()
+        {
+            TaskResources = ErrorString.ResourceManager;
+        }
+
+        #region Properties - Please keep these alphabetized.
+        public ITaskItem[] Analyzers
+        {
+            set
+            {
+                _store[nameof(Analyzers)] = value;
+            }
+
+            get
+            {
+                return _store[nameof(Analyzers)] as ITaskItem[];
+            }
+        }
+
+        public int CodePage
```

Comment:
Are all these properties legal and necessary in csi.exe? CodePage, especially, seems questionable.

I would rather not continue to talk about CodePages if we eventually want to be CoreCLR compatible.